src/overlays/actors/ovl_Obj_Toge/z_obj_toge.c

```diff
@@ -32,52 +35,277 @@ const ActorInit Obj_Toge_InitVars = {
+        func_800B2B44(globalCtx, &this->actor, &sp54, (Rand_ZeroOne() * 0.6f) + 2.8f);
+    }
+}
+
+void func_809A43A8(ObjToge* this, GlobalContext* globalCtx) {
+    if (!OBJTOGE_GET_4000(&this->actor)) {
+        func_809A41C0(this, globalCtx);
+    } else {
+        func_809A42A0(this, globalCtx);
+    }
+}
+
+s32 func_809A43EC(ObjToge* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    f32 temp_fv1 = player->actor.world.pos.x - this->unk_1B8;
+    f32 temp_fa0 = player->actor.world.pos.z - this->unk_1BC;
+    f32 temp = -(this->unk_1CC * temp_fa0) + (temp_fv1 * this->unk_1C8);
+    f32 temp2 = (this->unk_1CC * temp_fv1) + (temp_fa0 * this->unk_1C8);
+
+    return fabsf(temp) <= this->unk_1C0 && fabsf(temp2) <= this->unk_1C4;
+}
+
+void ObjToge_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjToge* this = THIS;
+    Path* path;
+    Vec3s* sp40;
+    s16 sp3E;
+    s32 sp38 = OBJTOGE_GET_4000(thisx);
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    Actor_SetScale(thisx, D_809A4D0C[sp38] * 0.1f);
+
+    sp3E = thisx->home.rot.z;
+    thisx->home.rot.z = thisx->world.rot.z = thisx->shape.rot.z = 0;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+
+    if (OBJTOGE_GET_PATH(thisx) == 0xFF) {
+        Actor_MarkForDeath(thisx);
+        return;
+    }
+
+    path = &globalCtx->setupPathList[OBJTOGE_GET_PATH(thisx)];
+    if (path->count != 2) {
+        Actor_MarkForDeath(thisx);
+        return;
+    }
+
+    sp40 = Lib_SegmentedToVirtual(path->points);
+    Math_Vec3s_ToVec3f(&this->unk_198[0], &sp40[0]);
+    Math_Vec3s_ToVec3f(&this->unk_198[1], &sp40[1]);
+    Math_Vec3f_Copy(&thisx->world.pos, &this->unk_198[0]);
+    thisx->world.rot.y = Math_Vec3f_Yaw(&this->unk_198[0], &this->unk_198[1]);
+    this->unk_194 = 0;
+    thisx->speedXZ = 0.0f;
+
+    if (sp3E > 0) {
+        s16 sp36;
+
+        this->unk_1B4 = true;
+        this->unk_1B8 = (this->unk_198[0].x + this->unk_198[1].x) * 0.5f;
+        this->unk_1BC = (this->unk_198[0].z + this->unk_198[1].z) * 0.5f;
+
+        sp36 = Math_Vec3f_Yaw(&this->unk_198[0], &this->unk_198[1]);
+        this->unk_1C8 = Math_CosS(sp36);
+        this->unk_1CC = Math_SinS(sp36);
+        this->unk_1C0 = sp3E * 7.5f;
+        this->unk_1C4 = Math3D_Distance(&this->unk_198[0], &this->unk_198[1]) * 0.5f;
+    } else {
+        this->unk_1B4 = false;
+    }
+
+    Collider_SetCylinder(globalCtx, &this->collider, thisx, &sCylinderInit);
+    Collider_UpdateCylinder(thisx, &this->collider);
+    thisx->colChkInfo.mass = MASS_IMMOVABLE;
+
+    if (sp38 == 1) {
+        this->collider.dim.radius = 60;
+        this->collider.dim.height = 40;
+    }
+
+    if (this->unk_1B4) {
+        func_809A4804(this);
+    } else {
+        func_809A4744(this);
+    }
+}
+
+void ObjToge_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    Collider_DestroyCylinder(globalCtx, &THIS->collider);
+}
```

Comment:
```suggestion
void ObjToge_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
    GlobalContext* globalCtx = globalCtx2;
    ObjToge* this = THIS;

    Collider_DestroyCylinder(globalCtx, &this->collider);
}
```
also matches and fits our style better.